src/test/java/uk/nhs/cdss/services/NarrativeServiceTest.java

```diff
@@ -0,0 +1,47 @@
+package uk.nhs.cdss.services;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.springframework.util.StringUtils.countOccurrencesOf;
+
+import java.util.List;
+import org.hl7.fhir.dstu3.model.Narrative.NarrativeStatus;
+import org.junit.Test;
+
+public class NarrativeServiceTest {
+
+  private final NarrativeService narrativeService = new NarrativeService();
+
+  @Test
+  public void buildNarrative_withNull_shouldBuildNullNarrative() {
+    var returnedNarrative = narrativeService.buildNarrative((String) null);
+
+    assertThat(returnedNarrative.getDivAsString(), nullValue());
+  }
+
+  @Test
+  public void buildNarrative_withText_shouldBuildNarrative() {
+    var returnedNarrative = narrativeService.buildNarrative("uniqueText");
+
+    assertThat(returnedNarrative.getDivAsString(), containsString("uniqueText"));
```

Comment:
There's a custom fhir matcher for narrative text
```suggestion
    assertThat(returnedNarrative, hasText("uniqueText"));
```